iterators_task_3/student/include/impl.h

```diff
@@ -14,9 +14,49 @@ struct pair_iterator : public Iterator
 template<class Iterator, class T>
 struct pair_iterator<Iterator, true, T> : public Iterator
 {
-    /** @todo Iterator traits */
-    /** @todo Iterator operations */
-    /** @todo Constructor from original iterator */
+    public:
+        /** @todo Iterator traits */
+        using value_type = const std::string;
+        using reference = typename Iterator::reference;
+        using pointer = typename Iterator::pointer;
+        using iterator_category = typename Iterator::iterator_category;
+        using difference_type = typename Iterator::difference_type;
+
+        /** @todo Iterator operations */
+        pair_iterator& operator++()
+        {
+            m_iter++;
+            return *this;
+        }
+
+        pair_iterator& operator--()
+        {
+            m_iter--;
+            return *this;
+        }
+
+        value_type& operator*()
+        {
+            return m_iter->first;
+        }
+
+        template<typename _T>
+        void operator=(const _T& val) = delete;
+        
+        operator value_type()
+        {
+            return m_iter->first;
+        }
+
+        /** @todo Constructor from original iterator */
+        pair_iterator() = default;
+
+        pair_iterator(Iterator it)
```

Comment:
I agree that it is better to adapt the interface, but here please just change the access type for that iterator